ui/page/root/wallet_selector_page.go

```diff
@@ -94,20 +163,40 @@ func (pg *WalletSelectorPage) HandleUserInteractions() {
 	pg.listLock.Lock()
 	defer pg.listLock.Unlock()
 
-	if ok, selectedItem := pg.walletComponents.ItemClicked(); ok {
-		pg.WL.SelectedWallet = pg.walletsList[selectedItem]
+	if ok, clickedItem := pg.walletComponents.ItemClicked(); ok {
+		tuple, exists := pg.indexMapping[clickedItem.Index]
```

Comment:
```suggestion
		tuple, exists := pg.indexMapping[clickedItem]
```
clicked item is already an index.. we dont need the index of an index.

What this here is doing is getting the index of selectedItems on the wallet selector, however, we already have the index here with `pg.indexMapping` which is same as what is on the wallet selector.